src/pluggy/_manager.py

```diff
@@ -181,7 +181,26 @@ def parse_hookimpl_opts(self, plugin: _Plugin, name: str) -> HookimplOpts | None
         customize how hook implementation are picked up. By default, returns the
         options for items decorated with :class:`HookimplMarker`.
         """
-        method: object = getattr(plugin, name)
+
+        # IMPORTANT: @property methods can have side effects, and are never hookimpl
```

Comment:
Given a similar recent change in pytest, maybe do it the same? https://github.com/pytest-dev/pytest/commit/c6a529032231ccddd3040e8ab1a5a756eb9ea4a0?

Reply:
If you look at the source of `safe_getattr`/`safe_isclass`, I believe it's actually the same: https://github.com/pytest-dev/pytest/blob/main/src/_pytest/compat.py#L273

Are you saying you'd prefer to import those funcs from pytest? Or copy/paste them directly?

Reply:
I've abstracted out the logic similarly to pytest in the latest commit, please review. It's simple enough that I don't think it warrants importing those functions from pytest, but it's your call.